src/main/java/games/coltexpress/ColtExpressGameState.java

```diff
@@ -199,10 +199,15 @@ protected AbstractGameState _copy(int playerId) {
     }
 
     @Override
-    protected double _getScore(int playerId) {
+    protected double _getHeuristicScore(int playerId) {
         return new ColtExpressHeuristic().evaluateState(this, playerId);
     }
 
+    @Override
+    public double getGameScore(int playerId) {
+        return getLoot(playerId).sumInt(Loot::getValue);
```

Comment:
(This comment is repeated in several places) - reflection in Java is very slow, and to be avoided if at all possible. It makes code look prettier, but we need it more to be fast, so I'd highly recommend changing all these to use old-school references to objects.

Reply:
I've looked into this in rather more detail, and run some experiments documented at https://github.com/GAIGResearch/TabletopGames/wiki/Coding-Conventions

The conclusion is that I think this is absolutely fine - there is no performance difference between lambda usage and method references. However I am going to avoid streams in highly core pieces of code (see the wiki link for my full argument).

Reply:
That looks like a great wiki page, good things to note, thanks! :)